### ISSUE TITLE
MPJPE from my own evaluating is inconsistent with that given in the paper

### ISSUE THREAD

**Issue by JihuaPeng** [paTRICK-swk/Pose3D-RIE#4]
Hello! Thank you for this great job! After I train the model according to the parameters you gave, I use the trained model to evaluate the MPJPE in the stage 3, but the results obtained are inconsistent with the results given in the paper. 
This is the MPJPE of my own training by using ground truth key points as input, but the MPJPE given in the paper is 30.1 mm.
![image](https://user-images.githubusercontent.com/30943144/133402529-da1989b7-9b1d-4fb1-8ec8-7f5769fa4f8f.png)
my pytorch version is 1.9.0, torchvision is 0.10.0,  batch size is 1024, channels is 256, checkpoint_frequency is 10, downsample is 1, dropout is 0.2, epochs is 80, latent_features_dim is 256, learing_rate is 0.001, lr_decay is 0.95.

If you have time, can you reply as soon as possible, thank you very much!

**Comment by paTRICK-swk** [comment]
Experiments were conducted in the following environment.
- OS: Ubuntu 20.04.1 LTS
- GPU: GeForce GTX 3080
- CUDA version: 11.1
- Pytorch version: 1.7.1

All these settings may affect the results. In addition, you also need to ensure that the seeds are the same at runtime.

**Comment by JihuaPeng** [comment]
> Experiments were conducted in the following environment.
> 
> * OS: Ubuntu 20.04.1 LTS
> * GPU: GeForce GTX 3080
> * CUDA version: 11.1
> * Pytorch version: 1.7.1
> 
> All these settings may affect the results. In addition, you also need to ensure that the seeds are the same at runtime.
Thanks for your reply! My seed is 4321, is your seed the same as mine?
And my channel dimension, dropout rate  and latten_feature of the TCN are set to 1024, 0.2, 256, respectively. My channel dimension, dropout rate  and latten_feature of the RIEModel are also set to 1024, 0.2, 256, respectively.

**Comment by paTRICK-swk** [comment]
The seed is the same. Make sure the settings of the networks are the same as those of `gt_pretrained.bin`. If the result is still not good, it might be caused by the difference between your experimental environment and the environment mentioned above.

**Comment by JihuaPeng** [comment]
Thanks for your reply! i will check it.

**Comment by JihuaPeng** [comment]
After I conducted the experiment of stage 2 and stage 3, I found that the MPJPE of the two steps is the same. I want to know if the command you gave (python run.py -k gt --stage 3 -lfd 256 -r stage_2_epoch_80.bin -lr 0.0005) did not conduct the fine-tuning network of stage 3, because the code enter the evaluation stage (Evaluating...) directly when stage 3 is conducted. In addition, I took a look at the pre-trained model about ground truth key points you gave , the number of epochs in it is 39, but the epochs of the fine-tuning network you gave on github is 20.

If you have time, can you reply as soon as possible, thank you very much!

**Comment by paTRICK-swk** [comment]
1) [This line](https://github.com/paTRICK-swk/Pose3D-RIE/blob/aaad25f617cbfa808ef3dcc99574b612cd3e1168/run.py#L260) of code ensures that the training process will not be skipped in the third stage, please make sure you did not delete it.
2) You can increase the number of epochs to get better performance, but basically, the results will not be much different.

**Comment by JihuaPeng** [comment]
Because your command in the stage 3 is 'python run.py -k gt --stage 3 -lfd 256 -r stage_2_epoch_80.bin -lr 0.0005'. -r means resume, so when the program runs the stage 3, it will conduct 290 lines and 291 lines of code, so the epoch is assigned the value 80 (the number of epochs saved in the stage 2). Therefore, when the code is conducted to the judgment of line 304, because the value of epoch is equal to the value of args.epoch, the following training process will not be conducted, and it will jump directly to the evaluation stage. I want to know what operations were performed in the fine-tuning phase of the third stage. Could you please tell me, thank you!

![image](https://user-images.githubusercontent.com/30943144/134136539-af7ee2b2-b33e-4674-bb48-20e7ae42bbb2.png)

**Comment by paTRICK-swk** [comment]
It is a bug, and I have fixed it. Thanks for pointing it out! Please check the latest code. In the third stage of optimization, the entire network will be trained for another 20 epochs for fine-tuning. The code snippet in your figure should not be run in this stage.

**Comment by JihuaPeng** [comment]
I want to know if you use the last saved model when you test, or the best model in training, because I train the network according to the code and environment you give, but I can’t gain the same results as you (30.1mm).😂

**Comment by paTRICK-swk** [comment]
If the performance is still poor, you can just use the pre-trained models we provide.
